### PR TITLE
Improve desktop/mouse scroll

### LINKED ISSUE #1079: Unnecessary log on zoom level changed
Every time map is resized in flutter WEB zoom level and some boolean is logged into console. How can I disable it?  
Log example:
```
false
zoom: 9.25
false
zoom: 9.5
false
zoom: 9.75
false
zoom: 10
false
zoom: 10.25
false
zoom: 10.5
false
zoom: 10.75
false
zoom: 11
false
zoom: 11.25
false
zoom: 11.5
false
zoom: 11.75
false
zoom: 12
false
zoom: 12.25
false
zoom: 12.5
false
zoom: 12.75
false
zoom: 12.5
false
zoom: 12.25
false
zoom: 12
false
zoom: 11.75
false
zoom: 11.5
false
zoom: 11.25
false
zoom: 11
false
zoom: 10.75
false
zoom: 10.5
false
zoom: 10.25
false
zoom: 10
```

### PR DESCRIPTION
**Pull request by marcin-jelenski**
The current mouse scroll is laggy.
Let's remove debounce method and use float zoom.

Tested on a physical mouse (Logitech) and MacBook touchpad.

ezgif-5-489115cf7e

Also fixed #1079

### CODE REVIEW

**Comment by JaffaKetchup** [comment]
Looking good! Haven't had much experience with this library on the web, but this looks nice from the .gif you added.

Just wondering out loud, could the removal of the debounce cause unintended side effects?

**Comment by marcin-jelenski** [comment]
@JaffaKetchup it seems debounce effect was introduced to limit number of zoom changes on scroll. Image loading is limited anyway by `TileLayerOptions::updateInterval`, so it was rather redundant. Also, the debounce method broke fling, which should work now.

**Comment by marcin-jelenski** [comment]
@kengu @johnpryan would you mind taking a look?

**Review by JaffaKetchup — APPROVED** [review]
LGTM, as long as this doesn't cause unnecessary processing stress. Just waiting on maintainers to return or people to be given write access. See #1117.

EDIT: Also see my comment below.

File: lib/src/gestures/gestures.dart
Comment (JaffaKetchup) [inline lib/src/gestures/gestures.dart:35]: @marcin-jelenski Just noticed this after I reviewed it. I'm not sure this is safe to assume (19 if null) - are there other locations in flutter_map where this assumption can be found?

File: lib/src/gestures/gestures.dart
Comment (marcin-jelenski) [inline lib/src/gestures/gestures.dart:35]: @JaffaKetchup there was no `?? 19` other check, so it was rather a mistake (copied from previous behaviour). Just changed this to coerce between `maxZoom` and infinity instead.

File: lib/src/gestures/gestures.dart
Comment (marcin-jelenski) [reply lib/src/gestures/gestures.dart]: @JaffaKetchup there was no `?? 19` other check, so it was rather a mistake (copied from previous behaviour). Just changed this to coerce between `maxZoom` and infinity instead.

**Review by JaffaKetchup — APPROVED** [review]
With those changes, LGTM I think.

**Comment by marcin-jelenski** [comment]
Thanks for the heads up @JaffaKetchup. 

We've decided to release our app with a fork attached (not to block release). You could test the new scroll behaviour here:
https://app.spacescripting.com/#/map

Fingers crossed for write access and getting this merged. It really improves the experience on desktop and web.

**Comment by JaffaKetchup** [comment]
No problem :).

Can't look at the website right now, as I'm on mobile and these changes don't affect the pinch gesture AFAIK.

Hoping for merge too!

**Comment by JaffaKetchup** [comment]
@stou Please pull upstream when possible, then I (or another maintainer) can merge after running the checks.

**Comment by marcin-jelenski** [comment]
@JaffaKetchup done, updated with upstream.

**Comment by JaffaKetchup** [comment]
Many thanks :)
Unfortunately, it looks like 'lib/src/gestures/gestures.dart' hasn't been formatted correctly, so the checks have failed. Please run `dart format` to fix the file, then I'll be ready to merge.
Thanks, and sorry for the hassle!

**Comment by marcin-jelenski** [comment]
@JaffaKetchup done - formatting fixed.

**Pull request merged**
